packages/lib-classifier/src/plugins/tasks/survey/components/components/Chooser/components/Choices/Choices.spec.js

```diff
@@ -39,78 +39,82 @@ describe('Component > Choices', function () {
     expect(choiceButtons[5]).to.have.text('Nothing here')
   })
 
-  it('should focus the next choice on arrow down keyDown', async function () {
+  it('should call handleDelete with choice ID on choice button backspace keyDown', async function () {
+    const handleDeleteSpy = sinon.spy()
     const user = userEvent.setup({ delay: null })
     render(
       <Grommet theme={zooTheme}>
         <Choices
           filteredChoiceIds={task.choicesOrder}
+          handleDelete={handleDeleteSpy}
+          focusedChoiceId='RDVRK'
           task={task}
         />
       </Grommet>
     )
-    await user.keyboard('{tab}')
     const choiceButtons = screen.getAllByRole('menuitemcheckbox')
     expect(choiceButtons[0]).to.equal(document.activeElement)
-    await user.keyboard('{arrowdown}')
-    expect(choiceButtons[1]).to.equal(document.activeElement)
+    await user.keyboard('{backspace}')
+    expect(handleDeleteSpy).to.have.been.calledOnceWith('RDVRK')
   })
 
-  it('should focus the previous choice on arrow up keyDown', async function () {
+  it('should call handleDelete with choice ID on choice button delete keyDown', async function () {
+    const handleDeleteSpy = sinon.spy()
     const user = userEvent.setup({ delay: null })
     render(
       <Grommet theme={zooTheme}>
         <Choices
           filteredChoiceIds={task.choicesOrder}
+          handleDelete={handleDeleteSpy}
+          focusedChoiceId='RDVRK'
           task={task}
         />
       </Grommet>
     )
-    await user.keyboard('{tab}')
     const choiceButtons = screen.getAllByRole('menuitemcheckbox')
     expect(choiceButtons[0]).to.equal(document.activeElement)
-    await user.keyboard('{arrowdown}')
-    expect(choiceButtons[1]).to.equal(document.activeElement)
-    await user.keyboard('{arrowup}')
-    expect(choiceButtons[0]).to.equal(document.activeElement)
+    await user.keyboard('{delete}')
+    expect(handleDeleteSpy).to.have.been.calledOnceWith('RDVRK')
   })
 
-  it('should call handleDelete with choice ID on choice button backspace keyDown', async function () {
-    const handleDeleteSpy = sinon.spy()
+  it('should call handleFocusedChoice with expected choice ID on arrowDown keyDown', async function () {
```

Comment:
Testing implementation like this is very brittle since it blocks you from refactoring your code. I know I’ve said this before, but this is an important lesson to learn. If you test the behaviour of your components then you can refactor safely, knowing that the behavioural tests will still pass if the refactor works.

Reply:
Behavioural tests for the menu here would be tests like:

- it should delete the selected choice on Backspace. 
- it should change the selected choice on ArrowUp. 
- it should change the selected choice on ArrowDown. 

Those behaviours won’t change when you refactor, even if you change the internal details of how state is stored and handled.

Reply:
I refactored the ArrowUp and ArrowDown tests in Choices accordingly, will refactor Backspace, Enter, and Space in #3816 .